source/extensions/filters/http/cache/cache_headers_utils.cc

```diff
@@ -0,0 +1,157 @@
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+std::ostream& operator<<(std::ostream& os, const OptionalDuration& duration) {
+  return duration.has_value() ? os << duration.value().count() : os << " ";
+}
+
+// TODO: clean or remove
+std::ostream& operator<<(std::ostream& os, const RequestCacheControl& request_cache_control) {
+  return os << "{"
+            << "must_validate: " << request_cache_control.must_validate << ", "
+            << "no_store: " << request_cache_control.no_store << ", "
+            << "no_transform: " << request_cache_control.no_transform << ", "
+            << "only_if_cached: " << request_cache_control.only_if_cached << ", "
+            << "max_age: " << request_cache_control.max_age << ", "
+            << "min_fresh: " << request_cache_control.min_fresh << ", "
+            << "max_stale: " << request_cache_control.max_stale << "}";
+}
+
+// TODO: clean or remove
+std::ostream& operator<<(std::ostream& os, const ResponseCacheControl& response_cache_control) {
+  return os << "{"
+            << "must_validate: " << response_cache_control.must_validate << ", "
+            << "no_store: " << response_cache_control.no_store << ", "
+            << "no_transform: " << response_cache_control.no_transform << ", "
+            << "no_stale: " << response_cache_control.no_stale << ", "
+            << "public: " << response_cache_control._public << ", "
+            << "max_age: " << response_cache_control.max_age << "}";
+}
+
+SystemTime CacheHeadersUtils::httpTime(const Http::HeaderEntry* header_entry) {
+  if (!header_entry) {
+    return {};
+  }
+  absl::Time time;
+  const std::string input(header_entry->value().getStringView());
+
+  // Acceptable Date/Time Formats per
+  // https://tools.ietf.org/html/rfc7231#section-7.1.1.1
+  //
+  // Sun, 06 Nov 1994 08:49:37 GMT    ; IMF-fixdate
+  // Sunday, 06-Nov-94 08:49:37 GMT   ; obsolete RFC 850 format
+  // Sun Nov  6 08:49:37 1994         ; ANSI C's asctime() format
+  static const auto& rfc7231_date_formats = *new std::array<std::string, 3>{
+      "%a, %d %b %Y %H:%M:%S GMT", "%A, %d-%b-%y %H:%M:%S GMT", "%a %b %e %H:%M:%S %Y"};
+  for (const std::string& format : rfc7231_date_formats) {
+    if (absl::ParseTime(format, input, &time, nullptr)) {
+      return ToChronoTime(time);
+    }
+  }
+  return {};
+}
+
+OptionalDuration CacheHeadersUtils::parseDuration(absl::string_view s) {
+  OptionalDuration duration;
+  // Strip quotation marks if any
+  if (!s.empty() && s.front() == '"' && s.back() == '"') {
+    s = s.substr(1, s.size() - 2);
+  }
+  long num;
+  if (absl::SimpleAtoi(s, &num) && num >= 0) {
+    // s is a valid string of digits representing a positive number
+    duration = std::chrono::seconds(num);
+  }
+  return duration;
+}
+
+std::tuple<absl::string_view, absl::string_view>
+CacheHeadersUtils::separateDirectiveAndArgument(absl::string_view full_directive) {
+  std::vector<absl::string_view> directive_parts =
```

Comment:
Whitespace is only allowed between full directives; it would be simpler to only StripAsciiWhitespace once (on full_directive). Also, if you change the return type of this function from tuple to pair, I think you can simplify the function body to:
{
  return absl::StrSplit(absl::StripAsciiWhitespace(full_directive), absl::MaxSplits('=', 1));
}

Reply:
Thanks for letting me know. Will do.

Reply:
I did not know absl::StrSplit could return an std::pair, great to know!